pkg/apiserver/handlers.go

```diff
@@ -72,13 +72,35 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+type LongRunningRequestCheck func(r *http.Request) bool
+
+// BasicLongRunningRequestCheck pathRegex operates against the url path, the queryParams match is case insensitive.
+// Any one match flags the request
+func BasicLongRunningRequestCheck(pathRegex *regexp.Regexp, queryParams map[string]string) LongRunningRequestCheck {
+	return func(r *http.Request) bool {
+		if pathRegex.MatchString(r.URL.Path) {
+			return true
+		}
+
+		for key, expectedValue := range queryParams {
+			for _, actualValue := range r.URL.Query()[key] {
```

Comment:
> Recommend not a loop, just use r.URL.Query().Get(key). Otherwise stuff like ?watch=true&watch=false will have surprising behavior, I think.

I'm fine with that, will update tomorrow.